File: src/main/java/cz/vhromada/catalog/gui/genre/GenreDataPanel.java
package cz.vhromada.catalog.gui.genre;

import javax.swing.GroupLayout;
import javax.swing.JLabel;

import cz.vhromada.catalog.facade.to.GenreTO;
import cz.vhromada.catalog.gui.commons.AbstractDataPanel;

/**
 * A class represents panel with genre's data.
 *
 * @author Vladimir Hromada
 */
public class GenreDataPanel extends AbstractDataPanel<GenreTO> {

    /**
     * SerialVersionUID
     */
    private static final long serialVersionUID = 1L;

    /**
     * Label for name
     */
    private JLabel nameLabel = new JLabel("Name");

    /**
     * Label with name
     */
    private JLabel nameData = new JLabel();

    /**
     * Creates a new instance of GenreDataPanel.
     *
     * @param genre TO for genre
     * @throws IllegalArgumentException if TO for genre is null
     */
    public GenreDataPanel(final GenreTO genre) {
        updateData(genre);

        initData(nameLabel, nameData);

        createLayout();
    }

    @Override
    protected void updateComponentData(final GenreTO data) {
        nameData.setText(data.getName());
    }

    @Override
    protected String getCzWikiUrl() {
        throw new IllegalStateException("Getting URL to czech Wikipedia page is not allowed for genres.");
    }

    @Override
    protected String getEnWikiUrl() {
        throw new IllegalStateException("Getting URL to english Wikipedia page is not allowed for genres.");
    }

    @Override
    protected String getCsfdUrl() {
        throw new IllegalStateException("Getting URL to ČSFD page is not allowed for genres.");
    }

    @Override
    protected int getImdbUrl() {
        throw new IllegalStateException("Getting URL to IMDB page is not allowed for genres.");
    }

    @Override
    protected GroupLayout.Group getHorizontalLayoutWithComponents(final GroupLayout layout, final GroupLayout.Group group) {
        return group
                .addGroup(createHorizontalDataComponents(layout, nameLabel, nameData));
    }

    @Override
    protected GroupLayout.Group getVerticalLayoutWithComponents(final GroupLayout layout, final GroupLayout.Group group) {
        return group
                .addGroup(createVerticalComponents(layout, nameLabel, nameData));
    }

}

File: src/main/kotlin/cz/vhromada/catalog/gui/genre/GenreDataPanel.kt
package cz.vhromada.catalog.gui.genre

import cz.vhromada.catalog.entity.Genre
import cz.vhromada.catalog.gui.common.AbstractDataPanel
import javax.swing.GroupLayout
import javax.swing.JLabel

/**
 * A class represents panel with genre's data.
 *
 * @author Vladimir Hromada
 */
class GenreDataPanel(genre: Genre) : AbstractDataPanel<Genre>() {

    /**
     * Label for name
     */
    private val nameLabel = JLabel("Name")

    /**
     * Label with name
     */
    private val nameData = JLabel()

    init {
        updateData(genre)

        initData(nameLabel, nameData)

        createLayout()
    }

    override fun updateComponentData(data: Genre) {
        nameData.text = data.name
    }

    override fun getCzWikiUrl(): String {
        error { "Getting URL to czech Wikipedia page is not allowed for genres." }
    }

    override fun getEnWikiUrl(): String {
        error { "Getting URL to english Wikipedia page is not allowed for genres." }
    }

    override fun getCsfdUrl(): String {
        error { "Getting URL to ČSFD page is not allowed for genres." }
    }

    override fun getImdbUrl(): Int {
        error { "Getting URL to IMDB page is not allowed for genres." }
    }

    override fun getHorizontalLayoutWithComponents(layout: GroupLayout, group: GroupLayout.Group): GroupLayout.Group {
        return group
                .addGroup(createHorizontalDataComponents(layout, nameLabel, nameData))
    }

    override fun getVerticalLayoutWithComponents(layout: GroupLayout, group: GroupLayout.Group): GroupLayout.Group {
        return group
                .addGroup(createVerticalComponents(layout, nameLabel, nameData))
    }

}
